src/cpp/rtps/reader/StatefulReader.cpp

```diff
@@ -77,6 +77,9 @@ bool StatefulReader::matched_writer_add(RemoteWriterAttributes& wdata)
 
     getRTPSParticipant()->createSenderResources(wdata.endpoint.remoteLocatorList, false);
 
+
+    wdata.endpoint.unicastLocatorList =
+        mp_RTPSParticipant->network_factory().ShrinkLocatorLists({wdata.endpoint.unicastLocatorList});
```

Comment:
Should we also create sender resources for unicast locators?

Should we make a similar change on StatelessReader?

Reply:
If I remember correctly StatelessReader doesn't create SenderResources. I make the minimal possible changes to this part of code.

Reply:
And I still think that sender resources should be created for the shrinked unicast locator list

Reply:
I had a conversation with @richiware the last week with these questions, and the conclusion was, leave as it is. Maybe you both can argument again and see the consequences. But please, share the conclusions!

Reply:
Ok. We will leave it as it is.